tf2_ros/test/test_buffer_client.cpp

```diff
@@ -66,6 +66,10 @@ class MockBufferServer : public rclcpp::Node
 
   void acceptedCallback(const std::shared_ptr<GoalHandle> goal_handle)
   {
+    // Simulate doing some work in here; otherwise, we can complete the goal
+    // before the rclcpp_action interface ever has time to do any work.
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Do you think this is a bug in rclcpp_action (ie. should it be reasonable to succeed/abort immediately)?

Reply:
Yeah, it's a good question.  I've gone back and forth on it in my head.  On the one hand, I think there is an argument that in a "real" action, there is always some delay between the action client send the goal and the action server getting it and then acting on it.  On the other hand, I can imagine a scenario where the client and server are composed together, and there is some kind of "fast-path" on the server that allows it to succeed immediately.

So I could go either way here; what's your thinking?

Reply:
The composition scenario also came to my mind. I'll take a closer look at the rclcpp_action implementation to see what could be done.